src/Symfony/Component/Clock/MockClock.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock;
+
+/**
+ * A clock that always returns the same date, suitable for testing time-sensitive logic.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+final class MockClock implements TimeInterface
+{
+    private \DateTimeImmutable $now;
+
+    public function __construct(\DateTimeImmutable|\DateTimeZone|string $now = null)
```

Comment:
Why don't you add another argument for the timezone? It would make the API cleaner & and this code simpler too.
Especially since we have named argument.

Reply:
I thought about it but it looks like those two arguments would be mutually exclusive. Passing both a TZ and a DateTime wouldn't make sense.

Reply:
but passing a **string** and a TZ make sense. 
use cases:
- User input in console / or form with `type=text`
- SGDB with poor date type like MySQL

I'm fine with `__construct(\DateTimeImmutable|\DateTimeZone $now = null)`

Reply:
Having an argument name that matches the expected value is more important than setting `null` in an argument. Having 2 arguments can be useful if the date part does not specify a timezone.

```suggestion
    public function __construct(\DateTimeImmutable|string $now = null, \DateTimeZone $timezone = null)
```

Usage:
```php
new MockClock('2022-06-20 18:20:00', 'Europe/Paris');
new MockClock(null, 'Europe/Paris');
```

Reply:
> but passing a string and a TZ make sense.

That convinced me :)
PR updated to split this in two arguments.